pkg/chaincode/approve.go

```diff
@@ -17,27 +17,15 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
-const approveTransactionName = "ApproveChaincodeDefinitionForMyOrg"
-
-type Definition struct {
-	Name                 string
-	Version              string
-	Sequence             int64
-	EndorsementPolicy    string
-	ValidationPlugin     string
-	ValidationParameters []byte
-	PackageID            string
-}
-
-// Install a chaincode package to specific peer.
-func Approve(ctx context.Context, connection grpc.ClientConnInterface, id identity.SigningIdentity, channelName string, chaincodeDef *Definition) error {
+// Approve a chaincode package to specific peer.
+func Approve(ctx context.Context, connection grpc.ClientConnInterface, id identity.SigningIdentity, chaincodeDef *Definition) error {
```

Comment:
I am still not convinced that `Definition` is the correct place for the `channelName` rather than being passed in as an explicit parameter. This is different to how Fabric itself models the chaincode definition

Reply:
may I have fabric  itself model?
but as the point we don't want to depends on fabric code. Hence do we have to keep same with fabric?